x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/VersionScriptDocValues.java

```diff
@@ -55,4 +56,63 @@ public String get(int index) {
     public int size() {
         return count;
     }
+
+    public boolean isValid() {
```

Comment:
`VersionScriptDocValues` is a list so I don't think this is the right level to apply the functions. We accept multiple values so there should be a way to apply these functions to the first, second or last entry in the list too ?
These functions could also be applied to any string (keyword family) field ? A static function like`semverIsRelease(String)` that would work on the decoded version could also be helpful imo, we don't need to put these functions under the new script values.